doc/source/development/rfc/rfc106_update_metadata.rst

```diff
@@ -0,0 +1,185 @@
+other reason, without checking the text of the error message)
+Some applications, e.g QGIS as discussed in https://github.com/qgis/QGIS/pull/60208
+have to use workarounds, such as having an allow-list of drivers known to work,
+and this is generally not desirable.
+This could also be used by the new "gdal raster edit" utility (https://gdal.org/en/latest/programs/gdal_raster_edit.html)
+to provide better error message when a user attempts to perform an operation
+that is not supported.
+
+Technical solution
+------------------
+
+A first addition is a GDAL_DCAP_UPDATE capability whose value must be set
+to "YES" when a driver offers some update capabilities.
+
+.. code-block:: c
+
+    /** Capability set by a driver that supports the GDAL_OF_UPDATE flag and offers
+     * at least some update capabilities.
+     * Exact update capabilities can be determined by the GDAL_DMD_UPDATE_ITEMS
+     * metadata item
+     * @since GDAL 3.11
+     */
+    #define GDAL_DCAP_UPDATE "DCAP_UPDATE"
+
+
+This offers the same type of capability discovery as the existing GDAL_DCAP_CREATE
+or GDAL_DCAP_CREATECOPY items.
+
+
+An additional item is provided to get more details on the exact capabilities:
+
+.. code-block:: c
+
+    /* List of (space separated) items that a dataset opened in update mode supports
+     * updating. Possible values are:
+     * - for raster: "GeoTransform" (through GDALDataset::SetGeoTransform),
+     *   "SRS" (GDALDataset::SetSpatialRef), "GCPs" (GDALDataset::SetGCPs()),
+     *   "NoData" (GDALRasterBand::SetNoDataValue),
+     *   "ColorInterpretation" (GDALRasterBand::SetColorInterpretation()),
+     *   "RasterValues" (GF_Write flag of GDALDataset::RasterIO() and GDALRasterBand::RasterIO()),
+     *   "DatasetMetadata" (GDALDataset::SetMetadata/SetMetadataItem), "BandMetadata"
+     *   (GDALRasterBand::SetMetadata/SetMetadataItem)
+     * - for vector: "Features" (OGRLayer::SetFeature()), "DatasetMetadata",
+     *   "LayerMetadata"
+     *
+     * No distinction is made if the update is done in the native format,
+     * or in a Persistent Auxiliary Metadata .aux.xml side car file.
+     *
+     * @since GDAL 3.11
+     */
+    #define GDAL_DMD_UPDATE_ITEMS "DMD_UPDATE_ITEMS"
+
+
+Note that the feature list is far from being exhaustive. It is restricted to
+what is thought to be the most useful.
+
+A new ``GDALDataset::ReportUpdateNotSupported(const char* pszDriverName)``
+static method will be added to avoid all drivers to have to compose their own
+error message when they don't support update.
+
+So current patterns like:
+
+.. code-block:: c++
+
+    if (poOpenInfo->eAccess == GA_Update)
+    {
+        CPLError(CE_Failure, CPLE_NotSupported,
+                 "The GSC driver does not support update access to existing "
+                 "datasets.");
+        return nullptr;
+    }
+
+will become:
+
+.. code-block:: c++
+
+    if (poOpenInfo->eAccess == GA_Update)
+    {
+        ReportUpdateNotSupported("GSC");
```

Comment:
At first read through I did not recognise that the string "GSC" was related to the calling driver.

Reply:
a good reason to kill all of them 🤣

Reply:
(temporarily pausing the troll in me) renaming to ReportUpdateNotSupportedByDriver()